dnsrecon.py

```diff
@@ -1431,69 +1432,183 @@ def main():
+            if len(answer) > 0:
+                # We will use the first IP found as the NS
+                ns_server.append(answer[0][2])
             else:
-                # Resolve in the case if FQDN
-                answer = socket_resolv(entry)
-                # Check we actually got a list
-                if len(answer) > 0:
-                    # We will use the first IP found as the NS
-                    ns_server.append(answer[0][2])
-                else:
-                    # Exit if we cannot resolve it
-                    print_error(f"Could not resolve NS server provided and server doesn't appear to be an IP: {entry}")
+                # Exit if we cannot resolve it
+                print_error(f"Could not resolve NS server provided and server doesn't appear to be an IP: {entry}")
 
         # User specified name servers but none of them validated
-        if len(ns_server) == 0:
+        if not ns_server:
             print_error('Please specify valid name servers.')
             sys.exit(1)
 
-    output_file = arguments.xml
-
-    if arguments.dictionary:
-        # print(arguments.dictionary)
-        if os.path.isfile(arguments.dictionary.strip()):
-            dictionary = arguments.dictionary.strip()
+        # ~ remove duplicated
+        ns_server = list(set(ns_server))
+
+    # ~ validating dictionary parameter
+    dictionary_required = []
+    if types:
+        # ~ combining the types and the type_map, we obtain
+        # ~ dictionary_required which is a list of bool
+        # ~ where True means that a dictionary file is required
+        dictionary_required = [type_map[t]['dictionary'] for t in types]
+
+    # ~ if a dictionary file is required,
+    # ~ we check that user has provided a valid one
+    if any(dictionary_required):
+        if arguments.dictionary:
+            d = arguments.dictionary.strip()
+            # ~ if the dictionary file provided by the user is valid,
+            # ~ we use it
+            if os.path.isfile(d):
+                dictionary = d
+            else:
+                print_error('The provided dictionary file does not exist!')
+                sys.exit(1)
         else:
-            print_error(f'File {arguments.dictionary.strip()} does not exist!')
-            exit(1)
+            # ~ otherwise we try to load a default dictionary file
+            print_status('No dictionary file has been specified.')
+            script_dir = os.path.dirname(os.path.realpath(__file__)) + os.sep
+            dictionaries = ['/etc/dnsrecon/namelist.txt', script_dir + 'namelist.txt']
+
+            dictionary = ""
+            for d in dictionaries:
+                if os.path.isfile(d):
+                    dictionary = d
+                    break
+
+            if not dictionary:
+                # ~ if we cannot find a valid dictionary file,
+                # ~ we exit
+                print_error("Cannot find the dictionary file provided with the tool.")
```

Comment:
Please use single quotes ever where it is applicable to use them to keep consistency with the quotes